components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -1481,4 +1493,40 @@ private String removeCRLFCharacters(String str) {
         return str;
     }
 
+    private void loadAndSaveAnalyticsSolutions(String solutionType) throws TemplateManagerServiceException {
+        if (Objects.nonNull(availableTemplateGroups.get(solutionType))) {
+            TemplateGroup templateGroup = availableTemplateGroups.get(solutionType);
+            List<RuleTemplate> ruleTemplates = templateGroup.getRuleTemplates();
+
+            for (RuleTemplate ruleTemplate : ruleTemplates) {
+                BusinessRuleFromTemplate businessRuleFromTemplate = replacePropertiesWithDefaultValues
+                        (ruleTemplate, templateGroup.getUuid());
+                try {
+                    createTablesIfNotExist();
+                    BusinessRule businessRule = queryExecutor.retrieveBusinessRule(ruleTemplate.getUuid());
+
+                    if (businessRule == null) {
+                        saveBusinessRuleDefinition(ruleTemplate.getUuid(), businessRuleFromTemplate,
+                                TemplateManagerConstants.DEPLOYED, 1);
+                        updateArtifactCount(ruleTemplate.getUuid(), 1);
+                    }
+                } catch (BusinessRulesDatasourceException | TemplateInstanceCountViolationException e) {
+                    throw new TemplateManagerServiceException("Saving APIM analytics feature business rule '" +
+                            businessRuleFromTemplate.getName() + "' to the database is failed. ", e);
+                }
+            }
+        }
+    }
+
+    private BusinessRuleFromTemplate replacePropertiesWithDefaultValues(RuleTemplate ruleTemplate,
+                                                                        String templateGroupUUID) {
+        Map<String, String> properties = new HashMap<>();
+        for (Map.Entry<String, RuleTemplateProperty> ruleTemplatePropertyEntry : ruleTemplate.getProperties()
```

Comment:
Try to stream values of `ruleTemplate.getProperties()` and collect to a Map.

Reply:
Fixed by b14dd18f396f799bd32670842869feb2475e778e